[c10d][CI] Improve world size setting in some tests

kwen2501 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138846

Following change in #137161 , bumping world size for some test suites.

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

kwen2501 committed:
[c10d][CI] Improve world size setting in some tests

[ghstack-poisoned]

fduwjj commented on test/distributed/test_c10d_object_collectives.py:
```diff
@@ -24,7 +24,6 @@
     sys.exit(0)
 
 BACKEND = dist.Backend.NCCL if torch.cuda.is_available() else dist.Backend.GLOO
-WORLD_SIZE = min(4, max(2, torch.cuda.device_count()))
```
what?!

kwen2501 commented:
@pytorchbot merge

cyyever commented:
@pytorchmergebot merge -f "Macos pending"

jeanschmidt commented:
@pytorchbot revert -m "introduced breaks in linux-focal-cuda11.8-py3.10-gcc9" -c nosignal

kwen2501 committed:
Update on "[c10d][CI] Improve world size setting in some tests"


Following change in #137161 , bumping world size for some test suites.

cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

kwen2501 commented:
@pytorchbot merge

kwen2501 commented:
@pytorchbot merge -f "The failed test is unrelated: docker: Error response from daemon"
